crates/router/src/core/routing/helpers.rs

```diff
@@ -543,3 +553,331 @@ pub fn get_default_config_key(
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("unable to retrieve success_rate configs")?;
+
+    let tenant_business_profile_id = format!(
+        "{}:{}",
+        state.tenant.redis_key_prefix,
+        business_profile.get_id().get_string_repr()
+    );
+
+    let success_based_connectors = client
+        .calculate_success_rate(
+            tenant_business_profile_id.clone(),
+            success_based_routing_configs.clone(),
+            routable_connectors.clone(),
+        )
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("unable to calculate/fetch success rate from dynamic routing service")?;
+
+    let payment_status_attribute =
+        get_desired_payment_status_for_success_routing_metrics(&payment_attempt.status);
+
+    let first_success_based_connector_label = &success_based_connectors
+        .labels_with_score
+        .first()
+        .ok_or(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable(
+            "unable to fetch the first connector from list of connectors obtained from dynamic routing service",
+        )?
+        .label
+        .to_string();
+
+    let (first_success_based_connector, merchant_connector_id) = first_success_based_connector_label
+        .split_once(':')
+        .ok_or(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable(
+            "unable to split connector_name and mca_id from the first connector obtained from dynamic routing service",
+        )?;
+
+    let outcome = get_success_based_metrics_outcome_for_payment(
+        &payment_status_attribute,
+        payment_connector.to_string(),
+        first_success_based_connector.to_string(),
+    );
+
+    core_metrics::DYNAMIC_SUCCESS_BASED_ROUTING.add(
+        &metrics::CONTEXT,
+        1,
+        &add_attributes([
+            ("tenant", state.tenant.name.clone()),
+            (
+                "merchant_id",
+                payment_attempt.merchant_id.get_string_repr().to_string(),
+            ),
+            (
+                "profile_id",
+                payment_attempt.profile_id.get_string_repr().to_string(),
+            ),
+            ("merchant_connector_id", merchant_connector_id.to_string()),
+            (
+                "payment_id",
+                payment_attempt.payment_id.get_string_repr().to_string(),
+            ),
+            (
+                "success_based_routing_connector",
+                first_success_based_connector.to_string(),
+            ),
+            ("payment_connector", payment_connector.to_string()),
+            (
+                "currency",
+                payment_attempt
+                    .currency
+                    .ok_or(errors::ApiErrorResponse::InternalServerError)
+                    .attach_printable("payment currency not found in payment_attempt")?
+                    .to_string(),
+            ),
```

Comment:
Won't currency be high cardinality data to be included as attributes in metrics?